03_recursion/java/04_sum/src/Sum.java

```diff
@@ -0,0 +1,20 @@
+import java.util.*;
+
+public class Sum {
+    public static int sum(ArrayList<Integer> num_list) {
+
+        if (num_list.size() == 0) {
+            return 0;
+        } else {
+            int num = num_list.get(0);
+            num_list.remove(0);
```

Comment:
could you write a version that doesn't modify the list?

Reply:
WIP